x/emissions/keeper/keeper.go

```diff
@@ -3410,3 +3439,125 @@ func (k *Keeper) GetPreviousForecasterScoreRatio(ctx context.Context, topicId To
 	}
 	return forecastTau, nil
 }
+
+// AddQualifiedInferer adds an inferer to the qualified inferers set for a topic
+func (k *Keeper) AddQualifiedInferer(ctx context.Context, topicId TopicId, inferer ActorId) error {
+	if err := types.ValidateTopicId(topicId); err != nil {
+		return errorsmod.Wrap(err, "invalid topic id")
+	}
+	if err := types.ValidateBech32(inferer); err != nil {
+		return errorsmod.Wrap(err, "invalid inferer address")
+	}
+	key := collections.Join(topicId, inferer)
+	return k.qualifiedInferers.Set(ctx, key)
+}
+
+// IsQualifiedInferer checks if an inferer is in the qualified inferers set for a topic
+func (k *Keeper) IsQualifiedInferer(ctx context.Context, topicId TopicId, inferer ActorId) (bool, error) {
+	key := collections.Join(topicId, inferer)
+	return k.qualifiedInferers.Has(ctx, key)
+}
+
+// RemoveQualifiedInferer removes an inferer from the qualified inferers set for a topic
+func (k *Keeper) RemoveQualifiedInferer(ctx context.Context, topicId TopicId, inferer ActorId) error {
+	key := collections.Join(topicId, inferer)
+	return k.qualifiedInferers.Remove(ctx, key)
+}
+
+// GetQualifiedInferersForTopic returns all qualified inferers for a specific topic
+func (k *Keeper) GetQualifiedInferersForTopic(ctx context.Context, topicId TopicId) ([]ActorId, error) {
+	var inferers []ActorId
+	rng := collections.NewPrefixedPairRange[TopicId, ActorId](topicId)
+	err := k.qualifiedInferers.Walk(ctx, rng, func(key collections.Pair[TopicId, ActorId]) (bool, error) {
+		inferers = append(inferers, key.K2())
+		return false, nil
+	})
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "error walking qualified inferers")
+	}
+	return inferers, nil
+}
+
+// AddQualifiedForecaster adds a forecaster to the qualified forecasters set for a topic
+func (k *Keeper) AddQualifiedForecaster(ctx context.Context, topicId TopicId, forecaster ActorId) error {
+	if err := types.ValidateTopicId(topicId); err != nil {
```

Comment:
This also should check the MaxForecastersToReward

Reply:
This is already checked in the unique function that calls `AddQualifiedForecaster `